src/components/InventoryDetail/DetailWrapper.js

```diff
@@ -13,17 +13,23 @@ import {
 } from '@patternfly/react-core';
 import PropTypes from 'prop-types';
 import { useSelector, useDispatch, useStore } from 'react-redux';
-import { toggleDrawer } from '../../store/actions';
+import { toggleDrawer, loadEntity } from '../../store/actions';
 import { BasicInfo, SystemIssues } from '../InventoryDetailDrawer';
 import FactsInfo from './FactsInfo';
 
-const DetailWrapper = ({ children, hideInvLink, showTags, Wrapper, className, hasAccess, appName, ...props }) => {
+const DetailWrapper = ({ children, hideInvLink, showTags, Wrapper, className, hasAccess, appName, inventoryId, ...props }) => {
     const dispatch = useDispatch();
     const store = useStore();
     const entity = useSelector(({ entityDetails }) => entityDetails?.entity);
     const isExpanded = useSelector(({ entityDetails }) => entityDetails?.isToggleOpened);
     const loaded = useSelector(({ entityDetails }) => entityDetails?.loaded);
 
+    useEffect(() => {
+        if (!entity || !(entity?.id === inventoryId) || !loaded) {
+            dispatch(loadEntity(inventoryId, { hasItems: true }, { showTags }));
+        }
+    }, []);
```

Comment:
What if app doesn't send `inventoryId`? Wouldn't this cause constant re-renders? Also, if  `entity` is changed via selector no new re-calculation is not triggered
```suggestion
    useEffect(() => {
        if (!entity || !(entity?.id === inventoryId) || !loaded) {
            dispatch(loadEntity(inventoryId, { hasItems: true }, { showTags }));
        }
    }, [entity, inventoryId, loaded]);
```

Reply:
I am just thinking about why the entity would change on this page. But, maybe there is something that I am not aware of.